File: app/src/main/java/com/lostincontext/commons/list/Adapter.java
package com.lostincontext.commons.list;


import android.support.v7.widget.RecyclerView;

public abstract class Adapter<VH extends ViewHolder> extends RecyclerView.Adapter<VH> {

    @Override public boolean onFailedToRecycleView(VH holder) {
        return holder.onFailedToRecycleView();
    }

    @Override public final void onBindViewHolder(VH holder, int position) {
        // let's not use that one
    }
}

File: app/src/main/java/com/lostincontext/commons/list/Adapter.kt
package com.lostincontext.commons.list


import android.support.v7.widget.RecyclerView

abstract class Adapter<VH : ViewHolder> : RecyclerView.Adapter<VH>() {

    override fun onFailedToRecycleView(holder: VH?): Boolean {
        return holder!!.onFailedToRecycleView()
    }

    override final fun onBindViewHolder(holder: VH, position: Int) {
        // no use
    }
}
